src/pydicom/filereader.py

```diff
@@ -674,6 +674,9 @@ def _not_group_0002(tag: BaseTag, vr: str | None, length: int) -> bool:
             fp, is_implicit_VR=False, is_little_endian=True, stop_when=_not_group_0002
         )
     )
+    file_meta.set_original_encoding(
+        is_implicit_vr=False, is_little_endian=True, character_encoding=default_encoding
+    )
```

Comment:
Can you not just pass `parent=default_encoding` to the `read_dataset()` call above?

Reply:
Fair enough!